rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+
+<a id="execution_info_id"></a>
+## Execution Info
+
+Policies are informed of key events through the reporting of Execution Info.
+Most commonly, this reporting is done by a backend and is not visible to the
+end user.  However, developers that implement custom backends, or that
+manage work submission without using the `submit` or `submit_and_wait` functions,
+will need to report Execution Info to allow policies to work properly. There
+are currently three kinds of Execution Info that may be required by a Policy:
+
+| Execution Info    | Value Type | Description |
+| ----------------- | ---------- | ----------- |
+| `task_time`       | `uint64_t` | Clock ticks that elapsed between task submission and completion. |
+| `task_submission` | void       | The task has been submitted. |
+| `task_completion` | void       | The task is complete |
+
+| Info Traits* | Description |
+| ------- | ----------- |
+| `report_info<S,Info>::value`, `report_info_v<S,Info>` | 'true' if the *Selection* requires the event type to be reported |
+| `report_value<S,Info,V>::value`, `report_info_v<S,Info,V>` | `true` if the *Selection* requires the event value to be reported |
+
+* These traits are defined in `include/oneapi/dpl/internal/dynamic_selection_traits.h`
+
+Backend traits can be used to determine what events are need by the *Policy* that provided a *Selection*.
+For example, below is code a function the receives a *Selection* and uses traits to determine if the
+`task_submission_t` must be reported. If so, it is reported using the `report` free function.
+
+```cpp
+    template <typename SelectionHandle, typename Function, typename... Args>
+    auto
+    submit(SelectionHandle s, Function&& f, Args&&... args)
+    {
+        constexpr bool report_task_submission = report_info_v<SelectionHandle, execution_info::task_submission_t>;
+        auto q = unwrap(s);
+        if constexpr (report_task_submission)
+            report(s, execution_info::task_submission);
+
+        // ... the remainder of the implementation
+
+    }
+```
+
+### Lazy Reporting 
+
+A backend may choose not to actively report events and instead lazily report them on demand by defining the
+optional `lazy_report` member function. A backend trait is provided to determine if calls to `lazy_report`
+are required.
+
+| Backend Traits* | Description |
+| ------- | ----------- |
+| `lazy_report`<B>::value`, `report_report_v<B>` | `true` if a *Backend* requires that a *Policy* calls to `lazy_report` before making a selection. |
+
+The example below shows how a `select` function might check this trait and call the function before it 
+applies its selection logic.
+
+```cpp
+    template <typename Function, typename... Args>
+    selection_type
+    select(Function&& f, Args&&... args)
+    {
+        if constexpr (backend_traits::lazy_report_v<Backend>)
+            backend_->lazy_report();
+
+        // rest of selection logic ....
+
+    }
+```
+
+<a id="exit_criteria_id"></a>
+## Exit Criteria
```

Comment:
Add an item about the oneDPL specification update to include the functionality.

Reply:
Will add.